### DISCUSSION TITLE
is there any specific reason that the editor's side panels need to automatically minimize?

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#28970] [Ideas]
Having these panels always visible doesn't feel like it's doing much, aside from being an inconvenience when you actually need to use them. The left panel can get away with being minimized, as there isn't much information in the expanded panel, but the right panel definitely feels like it should always be expanded
![osu_2024-07-20_19-45-24](https://github.com/user-attachments/assets/a14f9b7c-111d-4af8-b1d1-60efe9479420)

On 16:9 resolutions, I don't see any issue with just having them permanently expanded. On 4:3 resolutions, I can see there being value in the screen space gained, but I'm not sure how valuable designing around a 4:3 resolution is. [Steam's June 2024 hardware survey](https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam) shows that under 2.5% of people use a monitor at a 4:3 or 5:4 aspect ratio. Perhaps mapping in a 4:3 window is notably more popular? Lazer's flexibility in window scaling feels like it would be an adequate solution for that crowd regardless.

If the side panels need to have the ability to be minimized, then I would at least like to see a 'pin' feature, for those of us that do want them to always be visible.
![osu_2024-07-20_19-46-06](https://github.com/user-attachments/assets/2fb58005-a70c-4811-a2de-9b77ff7e7153)

**Comment by Stoppedpuma** [comment]
> [Steam's June 2024 hardware survey](https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam) shows that under 2.5% of people use a monitor at a 4:3 or 5:4 aspect ratio.

This is irrelevant as osu! has existed since when 4:3 was still common, there also seems to be a decent amount of players that still prefer to play using 4:3.

> Perhaps mapping in a 4:3 window is notably more popular?

The playfield is equal to 4:3.

**Reply by tovc** [reply]
>This is irrelevant as osu! has existed since when 4:3 was still common, there also seems to be a decent amount of players that still prefer to play using 4:3.

I'm not quite sure what you mean by this. Lazer has existed since 2017, and has only been the preferred client for about a year. Are you suggesting that there's some kind of technical limitation, or are you suggesting that we should be designing around the people that began playing in 2007, and somehow haven't upgraded their monitor since then?

>The playfield is equal to 4:3.

Okay, but that's the playfield and not the editor. There's no reason that the editor itself necessarily needs to stick to a 4:3 aspect ratio. I would prefer having, or having the option to have the to the side of the playfield not be dead space. I can understand the argument to have an editor that accommodates for 4:3 aspect ratios, but designing the editor around it, to the point where it hurts the notably more popular aspect ratio, just seems silly

To be perfectly clear, I *do* think that we should be designing with 4:3 aspect ratios in mind, especially with the advent of a mobile community that will largely consist of 4:3 tablets, but I don't think the current system is ideal for either use case. Having the ability to pin the panels is just my suggestion for something that can work better for desktop use. Tablets don't really have the idea of 'hovering', so the side panels feel incredibly obtuse on mobile

**Reply by Stoppedpuma** [reply]
> are you suggesting that we should be designing around the people that began playing in 2007, and somehow haven't upgraded their monitor since then?

No, I'm saying that just because steam says a percentage of people use something doesn't mean it's correct, especially in this case as it's a **hardware** survey, not what people are necessarily using through configuration. In the case of osu! there's several way to use 4:3, be it windowed mode, changing fullscreen resolution, screen scaling, etc.

> the side of the playfield not be dead space. I can understand the argument to have an editor that accommodates for 4:3 aspect ratios, but designing the editor around it, to the point where it hurts the notably more popular aspect ratio, just seems silly

How does designing the editor to accommodate for 4:3 ruin the experience for higher aspect ratios? You seem to be forgetting that there's other usages besides what's included inside the playfield, most obvious being storyboards, aspire map objects, and giving more horizontal space for taiko.

> Having the ability to pin the panels

I agree this could be useful but making the editor side panels fully expanded at all times just seems wrong as it takes up too much space.

**Reply by tovc** [reply]
>No, I'm saying that just because steam says a percentage of people use something doesn't mean it's correct, especially in this case as it's a hardware survey, not what people are necessarily using through configuration. In the case of osu! there's several way to use 4:3, be it windowed mode, changing fullscreen resolution, screen scaling, etc.

I'm aware of all of that. The steam hardware survey was simply to show that the vast majority of people aren't using a 4:3 monitor. Again, I'm well aware that windowed mode... exists, but I don't think it's so popular that we should be primarily designing around it. I don't think either of us have statistics regarding the preferred aspect ratio for mapping, so we'll drop this point.

>How does designing the editor to accommodate for 4:3 ruin the experience for higher aspect ratios? You seem to be forgetting that there's other usages besides what's included inside the playfield, most obvious being storyboards, aspire map objects, and giving more horizontal space for taiko.

I never said that it 'ruins the experience'. I only said that it makes actually using the side panels more inconvenient than it needs to be. If the space saved was actually used for something, then I can see the value in minimizing them, but the space that they occupy, on the vastly more popular aspect ratio, is dead space. It is being occupied by nothing.

Storyboarding is still not supported in lazer. These side panels are specific to placing circles, so there's no reason to have them visible in the storyboarding tab. Aspire maps just as often have objects off the top of the playfield. Should we also accommodate those? Peppy has stated that aspire maps are an 'edge case', and as such, support for them is limited. Neither of these seem like good reasons to have the side panels behave the way they do.

Taiko is honestly a reasonable argument against having the panels always be visible, but again, that's only one gamemode, and it has a fraction of the playerbase that stable has. I don't think that it alone should be the reason to have the side panels hidden most of the time.

>I agree this could be useful but making the editor side panels fully expanded at all times just seems wrong as it takes up too much space.

If you're talking about using a 4:3 aspect ratio, then sure, but on 16:9 and wider, the space that they occupy is being occupied by nothing. There is nothing to be gained by having them be hidden. It's just something that gets in the way when you actually want to use them.

**Reply by Stoppedpuma** [reply]
> I never said that it 'ruins the experience'.

Sorry, I should've clarified that was meaning like a question from your view less than a quote.

> Storyboarding is still not supported in lazer. These side panels are specific to placing circles, so there's no reason to have them visible in the storyboarding tab.

In my eyes it doesn't matter that something isn't supported yet as it's coming, there's not "no reason" to have them as the upcoming design tab will have the same side panels.

![image](https://github.com/user-attachments/assets/d636510e-0695-42ef-9ddf-1432b5114792)


> I don't think that it alone should be the reason to have the side panels hidden most of the time.

Even if it is one gamemode, it's still played by players and making it harder for mappers will only be negative. I'm also going to add that it doesn't matter "how small" the player base is, they still need a good experience as well. The recent changes towards the editor like transparency or making the timeline smaller were because of users saying the editor feels claustrophobic, I'm going to take a blind guess that changing this will likely cause complaints.

> There is nothing to be gained by having them be hidden. It's just something that gets in the way when you actually want to use them.

I wouldn't say there's nothing gained by having them be hidden as you yourself agreed with taiko, that's a reason no? Additionally since I rarely use the editor, isn't most of your time focused around objects and the playfield?

**Comment by peppy** [comment]
I agree there should be the option to keep them open. And maybe that should be default. Especially now that we have a lot of controls inside of them, it's easy for the thing-you're-doing to get pushed around vertically when the expansion happens, and it doesn't feel great.

**Reply by peppy** [reply]
https://github.com/ppy/osu/pull/30387

**Comment by Sanch-KK** [comment]
I actually think the opposite - that they take too much space even when collapsed. 4:3 mapping is a really popular thing amongst people who do that for a very long time (helps to feel the playfield better idk), and i cant help but feel that it feels very claustrophobic in that regard. Also sliders tend to go under that panel which is very annoying
![изображение](https://github.com/user-attachments/assets/f42761ea-41c3-46d5-a95e-7661ef97dca0)

There should be some option to squeeze it out even further imo, or to force toggle between the states

**Comment by cihe13375** [comment]
In stable I usually map with 4:3 mainly because it's easier to check offscreen objects in testplay. In lazer I don't do so because it already has a builtin offscreen checker (also it's hard to make an exact 4:3 window in lazer since the window size is arbitrary). Personally I don't feel much trouble from not using 4:3 in lazer (except maybe harder to reach the buttons in the right panel).

Having the side panel always expanded is definitely more convenient (and fixes the problem peppy said above which I agree with) though I think they are not the best in cosmetics --- the right side panel is horizontally too large (compared to the length of texts), and those big green sliders/buttons are grabbing too much attention. I can still live with it though, I think.

**Comment by cloudrac3r** [comment]
I would find it a lot easier to start learning mapping if the panels stayed expanded, because then I could read the button labels. When the panels are collapsed, they just show the abstract icons without labels, and I haven't learned what all the icons mean yet. If I could read the labels it would be a great help for starting out.

Currently the ≡ button toggles whether they stay open _vertically,_ but I haven't found this toggle to be very helpful - if I vertically collapse a section, it causes the layout to wobble around as it opens and closes. I think it would make more sense if the ≡ button kept the panels open _horizontally_ in the editor context.
